nuxeo-core/nuxeo-core-schema/src/main/java/org/nuxeo/ecm/core/schema/FacetNames.java

```diff
@@ -105,6 +105,14 @@
      */
     public static final String HAS_RELATED_TEXT = "HasRelatedText";
 
+    /**
+     * The document content will be stored as a cold (cool) storage, this is the case where the content represents:
+     * archiving, compliance...
```

Comment:
(cool) seems unneeded here, cold is enough no?
Also, I'd go for a sentence explaining that the document **can have** its content stored as cold storage, because if a document has this facet, it doesn't necessarily mean that its content **is** stored as cold storage.

Reply:
> (cool) seems unneeded here, cold is enough no?
I agree ;)

> Also, I'd go for a sentence explaining that the document can have its content stored as cold storage, because if a document has this facet, it doesn't necessarily mean that its content is stored as cold storage.

i was thinking that this facet was only for `cold storage` but if we say that is not automatically yes i will change it ;)

Reply:
That's a good question in fact, will it only be added dynamically when storing a content in cold storage? Or can we have some "cold storable content" documents, possibly stored in cold storage but not nevessarily?
@efge WDYT

Also, if at some point we decide that some content can be put back from cold to regular storage, then I guess that the facet won't be a synonym of "coldly stored"...

Reply:
@ataillefer @RSalem07 
My idea was to have it added dynamically only when a user/process decides to move to cold storage. But you're right that we could very well have doctypes that have this facet permanently. So the presence of the facet alone shouldn't be an indicator that something is in cold storage.

It's only when `coldstorage:coldContent` is not null then there's something in cold storage.

Reply:
Florent when you said `coldstorage:coldContent` is not null then there's something in cold storage. -> that means we should call `document.getPropertyValue(coldstorage:coldContent)` and i am wondering if this call will not throw an exception as we try to retrieve a cold storage or is there a thing that hold it to return not not value without trying to get the real content from cold storage 

thx

Reply:
From what I understand, if there is indeed a blob in this property, calling `getPropertyValue` will return a Blob, then we know that the content is in cold storage.
It is only when we try to get the binary content, by calling `getStream`, that the retrieval should happen, or an exception should be thrown, depending on how we want to handle the retrieval.
But @efge  will confirm.

Reply:
@ataillefer yes exactly. It's only when you call `blob.getStream()` that a request goes to S3 (and will fail if the blob retrieval has not been requested or is not finished or is expired)